modules/cudastereo/src/cuda/stereosgm_path_aggregation.cu

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2016 Fixstars Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "stereosgm_path_aggregation.hpp"
+/*
+#include "vertical_path_aggregation.hpp"
+#include "horizontal_path_aggregation.hpp"
+#include "oblique_path_aggregation.hpp"
+*/
+
+namespace cv { namespace cuda { namespace device
+{
+    namespace stereosgm
+    {
+        template <size_t MAX_DISPARITY>
+        void pathAggregation<MAX_DISPARITY>(const GpuMat& left, const GpuMat& right, GpuMat& dest, Stream& stream)
+        {
+            static const unsigned int NUM_PATHS = 8;
```

Comment:
Number of paths is fixed to 8 now.  
However, cv::StereoSGBM implementation has that default number of paths is 5 and it is changlable to full paths (8).

For a maintainability, we maybe have to add template parameter and specialization.